### ISSUE TITLE
Make post URLs more meaningful

### ISSUE THREAD

**Issue by kevinhikaruevans** [LemmyNet/lemmy-ui#2097]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

It would be nice to have post URLs that:
- have the community in the URL
- have a URL slug generated using the post's title

Whenever someone links a Lemmy post, I have no idea what the content will be from the URL. It would be nice if the URL gave clues to what to expect in the post.

Instead of having https://lemmy.example/post/1234, I would like to see https://lemmy.example/c/some-community/post/1234/this-is-some-thread (or something similar).

**Comment by ipkpjersi** [comment]
Yep, this would be a huge improvement. It would be a usability improvement and it would be an SEO improvement too.

**Comment by Die4Ever** [comment]
somewhat related, dealing with changing the post URL format

https://github.com/LemmyNet/lemmy/issues/2987#issuecomment-1690834945

it would probably be best to do these changes at the same time

**Comment by 8ullyMaguire** [comment]
I believe this is a backend issue and a duplicate of [Community name in post URL](https://github.com/LemmyNet/lemmy/issues/875)
